compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -957,6 +965,16 @@ impl<'a> Parser<'a> {
             }
         }
 
+        if self.token.kind == TokenKind::EqEq
+            && self.prev_token.is_ident()
+            && expected.iter().any(|tok| matches!(tok, TokenType::Token(TokenKind::Eq)))
+        {
+            // Likely typo: `=` → `==` in let expr or enum item
+            let mut err = self.sess.create_err(UseEqInstead { span: self.token.span });
+            err.emit();
```

Comment:
Don't emit the error here, it's being emitted later since you're returning it.

Reply:
Nice, thanks @compiler-errors.